src/Http/Response.php

```diff
@@ -2,175 +2,196 @@
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function withProtocolVersion($version) {
-    return $this->response->withProtocolVersion($version);
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getHeaders() {
-    return $this->response->getHeaders();
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function hasHeader($name) {
-    return $this->response->hasHeader($name);
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getHeader($name) {
-    return $this->response->getHeader($name);
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function getHeaderLine($name) {
-    return $this->response->getHeaderLine($name);
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function withHeader($name, $value) {
-    return $this->response->withHeader($name, $value);
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function withAddedHeader($name, $value) {
-    return $this->response->withAddedHeader($name, $value);
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function withoutHeader($name) {
-    return $this->response->withoutHeader($name);
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function withBody(StreamInterface $body) {
-    return $this->response->withBody($body);
-  }
+class Response implements ResponseInterface
+{
+    /**
+     * Response data.
+     *
+     * @var mixed
+     */
+    public $data;
+
+    /**
+     * Response instance.
+     *
+     * @var \Psr\Http\Message\ResponseInterface
+     */
+    protected $response;
+
+    /**
+     * Constructor.
+     *
+     * @param ResponseInterface $response
+     *   Response instance.
+     */
+    public function __construct(ResponseInterface $response)
+    {
+        $this->response = $response;
+        $this->data = $this->getDataFromResponse($response);
+    }
+
+    /**
+     * Get the API data from the response by name.
+     *
+     * @param string $name
+     *   Property name.
+     *
+     * @return mixed
+     *   Property data.
+     */
+    public function __get($name)
+    {
+        return $this->data->{$name};
+    }
+
+    /**
+     * Get the underlying data.
+     *
+     * @return mixed
+     *   Response data.
+     */
+    public function getData()
+    {
+        return $this->data;
+    }
+
+    /**
+     * Return an array of the data.
+     *
+     * @return array
+     *   Response data as array.
+     */
+    public function toArray()
+    {
+        return json_decode(json_encode($this->data, JSON_THROW_ON_ERROR), true, 512, JSON_THROW_ON_ERROR);
+    }
+
+    /**
+     * Get data from Response.
+     *
+     * @return mixed
+     *   Response data.
+     */
+    private function getDataFromResponse(ResponseInterface $response)
+    {
+        $contents = $response->getBody()->getContents();
+
+        return $contents ? json_decode($contents, null, 512, JSON_THROW_ON_ERROR) : null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBody(): StreamInterface
```

Comment:
Resolved the following issues in this class by adding return types to the methods below.

<img width="1472" alt="CleanShot 2023-06-21 at 21 43 58@2x" src="https://github.com/ChromaticHQ/orange-dam-php/assets/20355/4abb415f-39ba-4632-94d7-30f51d1faf37">